### PR TITLE
Fix for possible NullReferenceException

### PR DESCRIPTION
**Pull request by GeertvanHorrik**

### CODE REVIEW

**Comment by GeertvanHorrik** [comment]
Please discuss the code in the PR (Files changed tab) instead of the direct code tab, then it's easier to maintain.

File: Source/OxyPlot/Annotations/PathAnnotation.cs [hunk 1]
```diff
@@ -386,11 +386,14 @@ public override void Render(IRenderContext rc, PlotModel model)
         /// </returns>
         protected override HitTestResult HitTestOverride(HitTestArguments args)
         {
-            var nearestPoint = ScreenPointHelper.FindNearestPointOnPolyline(args.Point, this.screenPoints);
-            double dist = (args.Point - nearestPoint).Length;
-            if (dist < args.Tolerance)
+            if (this.screenPoints != null)
```
Comment (objorke) [inline Source/OxyPlot/Annotations/PathAnnotation.cs]: invert `if` statement to reduce nesting (Resharper recommends this, and I agree)

Comment (GeertvanHorrik) [inline Source/OxyPlot/Annotations/PathAnnotation.cs]: That means 2 return null, but ok

File: Source/OxyPlot/Rendering/ScreenPointHelper.cs [hunk 2]
```diff
@@ -22,11 +24,17 @@ public static class ScreenPointHelper
         /// <param name="point">The point.</param>
         /// <param name="points">The points.</param>
         /// <returns>The nearest point.</returns>
+        /// <exception cref="ArgumentNullException">The <paramref name="points"/> is <c>null</c>.</exception>
         public static ScreenPoint FindNearestPointOnPolyline(ScreenPoint point, IList<ScreenPoint> points)
         {
             double minimumDistance = double.MaxValue;
             var nearestPoint = default(ScreenPoint);
 
+            if (points == null)
```
Comment (objorke) [inline Source/OxyPlot/Rendering/ScreenPointHelper.cs]: isn't it normal to do the argument checks first? e.g. move this block to the top of the method?

Comment (objorke) [inline Source/OxyPlot/Rendering/ScreenPointHelper.cs]: sorry for commenting on the wrong place, I am still learning... thanks for the correction! :-)

File: Source/OxyPlot/Annotations/PathAnnotation.cs (hunk 1 above)
Comment (GeertvanHorrik) [inline Source/OxyPlot/Annotations/PathAnnotation.cs]: Resharper says this. is redundant, and I agree ;-)

Comment (objorke) [inline Source/OxyPlot/Annotations/PathAnnotation.cs]: right, but isn't nesting worse?

**Pull request merged**